### PR TITLE
Allow to use ON CLUSTER cluster_name in TRUNCATE syntax

### PR DESCRIPTION
**Pull request by git-hulk**
See https://clickhouse.com/docs/en/sql-reference/statements/truncate#truncate-table

### CODE REVIEW

**Comment by git-hulk** [comment]
cc @iffyio

File: src/parser/mod.rs [hunk 1]
```diff
@@ -708,13 +708,16 @@ impl<'a> Parser<'a> {
             };
         };
 
+        let on_cluster = self.parse_optional_on_cluster()?;
```
Comment (git-hulk) [inline src/parser/mod.rs:711]: ON CLUSTER is a ClickHouse-specific syntax, but we don't add the dialect guard for the sake of keeping consistent with other places.

**Review by iffyio — APPROVED** [review]
LGTM! cc @alamb

**Review by alamb — APPROVED** [review]
Thank yoU @iffyio  and @git-hulk

**Pull request merged**